File: applications/app-bemyplan/src/main/java/com/deploy/bemyplan/user/service/UserServiceUtils.java
package com.deploy.bemyplan.user.service;

import com.deploy.bemyplan.common.exception.model.ConflictException;
import com.deploy.bemyplan.domain.user.UserRepository;
import com.deploy.bemyplan.domain.user.UserSocialType;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class UserServiceUtils {

    static void validateNotExistsUserName(UserRepository userRepository, String name) {
        if (userRepository.existsByName(name)) {
            throw new ConflictException(String.format("이미 존재하는 닉네임 (%s) 입니다", name));
        }
    }

    static void validateNotExistsUser(UserRepository userRepository, String socialId, UserSocialType socialType) {
        if (userRepository.existsBySocialIdAndSocialType(socialId, socialType)) {
            throw new ConflictException(String.format("이미 존재하는 유저 (%s - %s) 입니다", socialId, socialType));
        }
    }
}

File: applications/app-bemyplan/src/main/kotlin/com/deploy/bemyplan/auth/application/UserServiceUtils.kt
package com.deploy.bemyplan.auth.application

import com.deploy.bemyplan.auth.application.port.out.GetUserPort
import com.deploy.bemyplan.common.exception.model.ConflictException
import com.deploy.bemyplan.domain.user.UserSocialType

internal fun validateNotExistsUserName(userRepository: GetUserPort, name: String) {
    if (userRepository.existsByName(name)) {
        throw ConflictException("이미 존재하는 닉네임 ${name} 입니다")
    }
}

internal fun validateNotExistsUser(userRepository: GetUserPort, socialId: String, socialType: UserSocialType) {
    if (userRepository.existsBySocialIdAndSocialType(socialId, socialType)) {
        throw ConflictException("이미 존재하는 유저 (${socialId} - ${socialType}) 입니다")
    }
}
